pegasus/helpers/hourofcode_helpers.rb

```diff
@@ -42,21 +42,21 @@ def hoc_canonicalized_i18n_path(uri, query_string)
     if HOC_I18N[possible_language]
       @user_language = possible_language
     else
-      path = File.join([possible_language, path].select {|i| !i.nil_or_empty?})
+      path = File.join([possible_language, path].reject(&:nil_or_empty?))
     end
   else
     @country = hoc_detect_country
-    path = File.join([possible_country_or_company, possible_language, path].select {|i| !i.nil_or_empty?})
+    path = File.join([possible_country_or_company, possible_language, path].reject(&:nil_or_empty?))
   end
 
   country_language = HOC_COUNTRIES[@country]['default_language']
   @language = @user_language || country_language || hoc_detect_language
```

Comment:
I'm _pretty sure_ `hoc_detect_language` is unreachable because `country_language` should always be present- not confident enough in this to change/remove this code though.